domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+
+  /** @var string[] Array of special-case policies for reassigning content. */
+  protected $reassignment_policies = ['prompt', 'default', 'ignore']; // + machine name;
+
+  /**
+   * Get a domain storage object or throw an exception.
+   *
+   * @return \Drupal\domain\DomainStorageInterface
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  protected function getStorage() {
+    if (!is_null($this->domain_storage)) {
+      return $this->domain_storage;
+    }
+
+    try {
+      $this->domain_storage = \Drupal::entityTypeManager()->getStorage('domain');
+    }
+    catch (PluginNotFoundException $e) {
+      throw new DomainCommandException('Unable to get domain: no storage', $e);
+    }
+    catch (InvalidPluginDefinitionException $e) {
+      throw new DomainCommandException('Unable to get domain: bad storage', $e);
+    }
+
+    return $this->domain_storage;
+  }
+
+  /**
+   * Lookup a string identifying a domain and return the object, or throw error.
+   * 
+   * @param string $argument
+   *   The machine name, or the hostname, of an existing domain.
+   *
+   * @return \Drupal\domain\DomainInterface
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  protected function getDomainFromArgument($argument) {
+    $domain_storage = $this->getStorage();
+
+    // Try loading domain assuming arg is a machine name.
+    $domain = $domain_storage->load($argument);
+    if (!$domain) {
+      // Try loading assuming it is a host name.
+      $domain = $domain_storage->loadByHostname($argument);
+    }
+
+    // domain_id (an INT) is only used internally because the Node Access
+    // system demands the use of numeric keys. It should never be used to load
+    // or identify domain records. Use the machine_name or hostname instead.
+    if (!$domain) {
+      throw new DomainCommandException(
+        dt('Domain record could not be found from "!a".', ['!a' => $argument])
+      );
+    }
+
+    return $domain;
+  }
+
+  /**
+   * Filter a list of domains by excluding domains appearing in a specific list.
+   *
+   * @param DomainInterface[] $domains
+   *   List of domains.
+   * @param int[] $exclude
+   *   List of domain names to exclude from the list.
+   * @param DomainInterface[] $initial
+   *   Initial value of list that will be returned.
+   *
+   * @return array
+   */
+  protected function filterDomains($domains, $exclude, $initial = []) {
+    foreach ($domains as $one) {
+      // Exclude unwanted domains.
+      if (!in_array($one->id(), $exclude)) {
+        $initial[$one->id()] = $one;
+      }
+    }
+    return $initial;
+  }
+
+  /**
+   * Check the domain response.
+   *
+   * @param \Drupal\domain\DomainInterface $domain
+   *
+   * @return string
+   */
+  protected function checkHTTPResponse(DomainInterface $domain, $validate_url = FALSE) {
+    // Ensure the url is rebuilt.
+    if ($validate_url) {
+      $this->validateDomain($domain->getHostname());
```

Comment:
We want to use `DomainValidator::checkResponse()` here, not the full validation routine in `DomainValidator::validate()`

Reply:
Ok, I have adapted that code. not sure the new version is correct as I realized this code was expected to return something, and it wasn't. Now returning TRUE if asked to validate and the response code is 2xx, or not validating.